RM_SUI/RM_SUI/View/SignInView.swift

```diff
@@ -0,0 +1,180 @@
+                        .focused($isPasswordTextFieldFocused)
+                    Divider()
+                        .padding(.horizontal, Constants.horizontalPadding)
+                    signUpLogInButtonView
+                    forgotPasswordButtonView
+                    checkVerification
+                }
+                .animation(.easeInOut)
+                .navigationBarBackButtonHidden()
+            }
+            .onTapGesture {
+                isPhoneTextFieldFocused = false
+                isPasswordTextFieldFocused = false
+            }
+        }
+        .navigationBarHidden(true)
+    }
+
+    // MARK: - Private Properties
+
+    private let formatter = PhoneNumberFormatter()
+
+    @FocusState private var isPhoneTextFieldFocused: Bool
+    @FocusState private var isPasswordTextFieldFocused: Bool
+
+    @StateObject private var signInViewModel = SignInViewModel()
+
+    private var backgroundGradientView: some View {
+        VStack {
+            LinearGradient(
+                colors: Constants.backgroundGradient,
+                startPoint: .leading,
+                endPoint: .trailing
+            )
+            .ignoresSafeArea(.all)
+            .frame(maxHeight: 0)
+            Spacer()
+        }
+    }
+
+    private var signUpLogInToggleView: some View {
+        HStack {
+            Button {
+                signInViewModel.isSignUp = false
+            } label: {
+                Text(Constants.loginTextName)
+                    .fontWeight(.bold)
+                    .foregroundColor(.purple)
+            }
+            .padding(Constants.buttonPadding)
+            .frame(maxWidth: .infinity)
+            .background(signInViewModel.isSignUp ? Color(.systemBackground) : Color(.systemGray5))
+
+            Button {
+                signInViewModel.isSignUp = true
+            } label: {
+                Text(Constants.signUpTextName)
+                    .fontWeight(.bold)
+                    .foregroundColor(.purple)
+            }
+            .padding(Constants.buttonPadding)
+            .frame(maxWidth: .infinity)
+            .background(signInViewModel.isSignUp ? Color(.systemGray5) : Color(.systemBackground))
+        }
+        .font(.largeTitle)
+        .overlay(content: {
+            Capsule()
+                .stroke(Color(.systemGray5), lineWidth: Constants.capsuleStrokeLineWidth)
+        })
+        .mask { Capsule() }
+        .padding(Constants.hStackPadding)
+    }
+
+    private var phoneTextFieldView: some View {
+        TextField(Constants.numberPlaceholderTextName, text: $signInViewModel.phoneNumber)
+            .onChange(of: signInViewModel.phoneNumber, perform: { newValue in
+                signInViewModel.phoneNumber = formatter.string(for: newValue) ?? ""
+                if signInViewModel.phoneNumber.count == Constants.phoneNumberCount {
+                    isPhoneTextFieldFocused = false
+                    isPasswordTextFieldFocused = true
+                }
```

Comment:
```suggestion
            .onChange(
            viewModel.formatPhoneNumber(text: text)
```
   можно посмотреть у Эрнеста